server/py/framework/utils/singletons/k8s.py

```diff
@@ -604,6 +604,8 @@ def read_secret_data(
         self, secret_name: str, namespace: str = "", load_as_json=False, silent=False
     ) -> dict[str, str]:
         k8s_secret = self.read_secret(secret_name, namespace, silent)
+        if not k8s_secret:
+            return {}
```

Comment:
if no k8s_secret then return `None`
empty dict means empty secret - but if there's no secret, return None should happen as it tells the callee more about what happened